src/Raven.Server/Utils/CommandLineSwitches.cs

```diff
@@ -2,28 +2,135 @@
 
-        public static string[] ParseAndRemove(string[] args)
+        public static bool RunAsService => ParseSwitchOption(_runAsServiceOption);
+
+        public static bool RegisterService => ParseSwitchOption(_registerServiceOption);
+
+        public static bool UnregisterService => ParseSwitchOption(_unregisterServiceOption);
+
+        public const string DefaultServiceName = "RavenDB";
+
+        public static string ServiceName => 
+            _serviceNameOption.Values.FirstOrDefault() ?? DefaultServiceName;
+
+        public static string CustomConfigPath =>
+            _customConfigPathOption.Values.FirstOrDefault();
+
+        public static bool ShouldShowHelp => ParseSwitchOption(_helpOption);
+
+        private static CommandLineApplication _app;
+
+        private static CommandOption _printIdOption;
+
+        private static CommandOption _browserOption;
+
+        private static CommandOption _versionOption;
+
+        private static CommandOption _registerServiceOption;
+
+        private static CommandOption _unregisterServiceOption;
+
+        private static CommandOption _runAsServiceOption;
+
+        private static CommandOption _serviceNameOption;
+
+        private static CommandOption _helpOption;
+
+        private static CommandOption _customConfigPathOption;
+
+        public static void ShowHelp()
+        {
+            _app.ShowHelp();
+        }
+
+        public static string[] Process(string[] args)
         {
             if (args == null)
                 return null;
 
-            var list = args.ToList();
-            PrintServerId = list.Remove("--print-id");
-            LaunchBrowser = list.Remove("--browser");
-            PrintVersionAndExit = list.Remove("-v") || list.Remove("--version");
-            return list.ToArray();
+            var nonConfigurationSwitches = args
+                .Where(x => Regex.IsMatch(x, @"^(--|\/)[rR]aven") == false)
+                .ToArray();
+
+            _app = new CommandLineApplication();
+            _helpOption = _app.Option(
+                "-h | -? | --help",
+                "Shows help",
+                CommandOptionType.NoValue);
+            _versionOption = _app.Option(
+                "-v | --version",
+                "Displays version and exits",
+                CommandOptionType.NoValue);
+            _printIdOption = _app.Option(
+                "--print-id", 
+                "Prints server ID upon server start", 
+                CommandOptionType.NoValue);
+            _registerServiceOption = _app.Option(
+                "--register-service",
+                "Registers Windows service and exits",
+                CommandOptionType.NoValue);
+            _unregisterServiceOption = _app.Option(
+                "--unregister-service",
+                "Unregisters Windows service and exits",
+                CommandOptionType.NoValue);
+            _runAsServiceOption = _app.Option(
+                "--run-as-service",
```

Comment:
Yeah, you're right `--daemon` would fit better here. On Windows we just have `--register-service` and `--unregister-service` (then manage it from `services.msc`) and on Linux we have a daemon mode - process is waiting.

I blocked this on Windows to avoid confusing the user - why 2 service modes etc.